lib/Video/Adapter/Ffmpeg.php

```diff
@@ -416,6 +446,34 @@ public function scaleByHeight($height)
     {
         // ensure $height is even (mp4 requires this)
         $height = ceil($height / 2) * 2;
-        $this->addArgument('-filter:v', 'scale=trunc(oh/(ih/iw)/2)*2:'.$height);
+        $this->videoFilter[] = 'scale=trunc(oh/(ih/iw)/2)*2:'.$height;
+    }
+
+    /**
+     * @param string|null $inputSeeking
+     * @param string|null $targetDuration
+     */
+    public function cut(?string $inputSeeking = null, ?string $targetDuration = null): void
+    {
+        if (!empty($inputSeeking)) {
+            $parts = explode(':', $inputSeeking);
+            $this->inputSeeking = (float) ((int) $parts[0] * 3600 + (int) $parts[1] * 60 + (float) $parts[2]);
```

Comment:
@kingjia90 the reason behind it is that this way we are able to do some calculations with the time, for example in this implementation it will reset the start position to 0 if it is beyond the duration of the video. In the future we can reset it not to 0 but recalculate the position as `start = sourceDuration - targetDuration`. All of above is easier to do if all the values are numerical.

Reply:
ah i see, that could be a valid point. 
Can we do the similar "validation" and conversion as here https://github.com/pimcore/pimcore/pull/12449/files, so the values would be "compatible" with each other.

Reply:
Ok, I've changed the conversion, the "validation" might follow. The one for duration is limited to 99 hours, as it only checks two digits per group. For the same reason it needs seconds with leading zero. Not that it is a big issue. But hey, now there is a color chanel mixer like described in [https://hhsprings.bitbucket.io/docs/programming/examples/ffmpeg/manipulating_video_colors/colorchannelmixer.html](https://hhsprings.bitbucket.io/docs/programming/examples/ffmpeg/manipulating_video_colors/colorchannelmixer.html). With presets to choose from or, if you know what you are doing, custom string. At this point I need to update the docs.

Reply:
Thank you, but for validation i meant the `preg_match` instead of the `explode` and eventually aborting the video thumbnail generation process when returning `false`.

About the color channel mixer, that is cool and can see it will come in handy, but it need to be somehow extendable so devs can add more presets or manually define own custom values 🤔, and please rename the PR title accordigly.

Reply:
@kingjia90 - ok, preg_match it will be. The "effect" setting for the colormixer is in fact a combobox, so you can just write a custom value into it OR select a preset. The preview feature needs to be added for this one to be more usefull (like the one you mentioned with downloadable thumbnails). Lots to do indeed.